paimon-core/src/main/java/org/apache/paimon/append/AppendOnlyWriter.java

```diff
@@ -63,12 +69,17 @@ public class AppendOnlyWriter implements RecordWriter<InternalRow> {
     private final List<DataFileMeta> compactAfter;
     private final LongCounter seqNumCounter;
     private final String fileCompression;
+    private final boolean useWriteBuffer;
+    private final boolean spillable;
     private final FieldStatsCollector.Factory[] statsCollectors;
+    private final IOManager ioManager;
 
+    private SpillableBuffer writeBuffer;
```

Comment:
You can create a small inner `interface`. And with two small inner class implement it.

Reply:
done